src/core/lib/event_engine/event_engine_factory.cc

```diff
@@ -36,12 +36,18 @@ void SetDefaultEventEngineFactory(
   g_event_engine_factory = factory;
 }
 
-std::unique_ptr<EventEngine> CreateEventEngine() {
+void MaybeSetDefaultEventEngineFactory(
```

Comment:
Should this return a boolean? It's a bit odd that this function might do nothing and not inform its caller.

Reply:
I'm not sure what the caller (`grpc_init`) would do with that information. If anything, I believe that failure to initialize the EventEngine factory should abort the process early, otherwise things will go badly in less obvious ways.

Reply:
@markdroth If an application developer wants to use a custom EventEngine of their own design, they are expected to call `SetDefaultEventEngineFactory` before grpc_init is called. We'd be overwriting their custom factory if we didn't check first.

Reply:
I suggest you inline this function into InitEventEngineFactory. It has no other callers, and its name is misleading; The word `Default` appears in it, but its implementation lacks any knowledge of defaults. I think the intent would be much easier to understand were it inlined.

Reply:
We have not yet determined what we want our long-term story for registering a global EventEngine impl to be, so I don't think we should try to solve that problem at this point.  Right now, we only need to address first-party EventEngine impls.

Reply:
> [SetDefaultEventEngineFactory] has no other callers

It is part of the public EventEngine API. `Default` may be a bit misleading, I'm open to naming alternatives, but `default` is in contrast to those EventEngine objects that users can provide manually on a channel or server creation. If none are provided, a one will be created using the default factory.
 https://github.com/grpc/grpc/blob/e523825c75aba052a3e36362fc20cdb9abca0b8d/include/grpc/event_engine/event_engine.h#L381-L391.

Reply:
I'd feel more comfortable if:
- default and user-provided factories were held in different variables
- this method set the default
- this method (and perhaps the user-provided setter) aborted the process if no action is taken or returned sufficient information to allow the caller to do so

Reply:
> I'd feel more comfortable if default and user-provided factories were held in different variables

Why? There is no use case where we need both a user-provided factory and a gRPC-provided built-in factory. If users supply their own EventEngines and factories, gRPC should not use anything else. Unless there's a new requirement to use both concurrently (which could become a thing, I suppose), I think it's easier to prevent gRPC from using an unwanted factory if it can't find it to begin with.

> I'd feel more comfortable if this method (and perhaps the user-provided setter) aborted the process if no action is taken or returned sufficient information to allow the caller to do so.

I don't understand your suggestion, can you try to explain further? This method sets a pointer and does nothing else. I imagine it would be friendly to abort the process if it's the nullptr, but otherwise I'm not sure what kind of validation would be meaningful at this point.

Reply:
Ah, I think I misunderstood the meaning of `Default` in this context.

A user application that sets `grpc_eventengine_strategy` *and* calls `SetDefaultEventEngineFactory` seems like it is misconfigured? Should we complain about that?

Also, are users expected to call `MaybeSetDefault...`? If not, should it be part of the public API?

Reply:
> A user application that sets grpc_eventengine_strategy and calls SetDefaultEventEngineFactory seems like it is misconfigured? Should we complain about that?

Good point, probably. Mark suggested we remove the documentation around `grpc_eventengine_strategy` because it isn't ready for public consumption. This is another reason to keep it undocumented for now. I'll add a code comment indicating it's for internal usage at the moment.

> Also, are users expected to call `MaybeSetDefault...`? 

No

> If not, should it be part of the public API?

It's not. `src/core/lib/event_engine/event_engine_factory.h` is an internal file. Public interfaces are declared in `include/grpc/event_engine/...`